File: Python/Bucles_while/Avanzado/Ejercicio_02.py
# Inicialización de variables
intento = 0

# Asignar un número aleatorio al número a adivinar
numero_aleatorio = 50  # Puedes elegir cualquier número en el rango de 1 a 100

# Ciclo para adivinar el número
while True:
    # Incrementar el contador de intentos
    intento += 1

    # Solicitar al usuario adivinar el número
    numero_adivinanza = int(input(f"Intento {intento}: Ingrese un número entre 1 y 100: "))

    # Verificar si el número es igual, mayor o menor
    if numero_adivinanza == numero_aleatorio:
        print("¡Felicitaciones! Has adivinado el número correctamente.")
        break
    elif numero_adivinanza < numero_aleatorio:
        print("El número es mayor. Intenta nuevamente.")
    else:
        print("El número es menor. Intenta nuevamente.")

File: Java/src/main/java/org/sisuns/bucles_while/avanzado/Ejercicio_02.java
package org.sisuns.bucles_while.avanzado;

public class Ejercicio_02 {
    public static void main(String[] args) {
        // Declaración de variables
        int numeroAleatorio, intento = 0, numeroAdivinanza;

        // Inicialización de variables
        // Puedes cambiar el número aleatorio según tus preferencias
        numeroAleatorio = 50;

        // Ciclo para adivinar el número
        do {
            // Incrementar el contador de intentos
            intento++;

            // Solicitar al usuario adivinar el número
            System.out.print("Intento " + intento + ": Ingrese un número entre 1 y 100: ");
            numeroAdivinanza = Integer.parseInt(System.console().readLine());

            // Verificar si el número es igual, mayor o menor
            if (numeroAdivinanza == numeroAleatorio) {
                System.out.println("¡Felicitaciones! Has adivinado el número correctamente.");
            } else {
                if (numeroAdivinanza < numeroAleatorio) {
                    System.out.println("El número es mayor. Intenta nuevamente.");
                } else {
                    System.out.println("El número es menor. Intenta nuevamente.");
                }
            }

        } while (numeroAdivinanza != numeroAleatorio);
    }
}
